client/jetpack-onboarding/main.jsx

```diff
@@ -186,14 +186,21 @@ export default connect(
 		).isLoading;
 
 		const userIdHashed = getUnconnectedSiteUserHash( state, siteId );
+
+		// Only show the Stats Step either if we aren't connected to WP.com yet,
+		// or if we're just being redirected back to JP Onboarding right after
+		// going through JP Connect, in which case the `action` query arg will be
+		// set to `activate_stats`.
+		const showStatsStep = ! isConnected || action === 'activate_stats';
```

Comment:
Right now, for connected sites we'll only show this step only if we have the `activate_stats` action in the route. This is fine, but do you think it makes sense to take this a step further and actually use the real module activation state? Because right now, we might skip this step for connected sites, regardless of whether the module is active or not.

Reply:
That would mean changing the `isConnected` criterion tho, not the `action === ...` one, right? I guess we can do that, it just requires yet another query component, selector,... 🙄 Maybe keep the  `isConnected` criterion? 😂Can users even disable the stats module once connected (other than through the CLI)?

Reply:
Yeah, you're right that we're probably going too far with this. Maybe keep it simple is a good idea 👍